src/yoroi-wallets/cardano/shelley-testnet/constants.ts

```diff
@@ -0,0 +1,140 @@
+import {DefaultAsset, TokenInfo} from '../../types'
+import {CardanoHaskellShelleyNetwork} from '..'
+export * from '../../cardano/constants'
+
+export const NETWORK_ID = 300
+export const WALLET_IMPLEMENTATION_ID = 'haskell-shelley'
+export const API_ROOT = 'https://preprod-backend.yoroiwallet.com/api'
+export const TOKEN_INFO_SERVICE = 'https://metadata.cardano-testnet.iohkdev.io'
+export const DISCOVERY_GAP_SIZE = 20
+export const DISCOVERY_BLOCK_SIZE = 50 // should be less than API limitations
+export const MAX_GENERATED_UNUSED = 20 // must be <= gap size
+export const PURPOSE = 2147485500
+export const COIN_TYPE = 2147485463
+export const ACCOUNT_INDEX = 0
+export const HARD_DERIVATION_START = 2147483648
+export const CHIMERIC_ACCOUNT = 2
+export const STAKING_KEY_INDEX = 0
+export const STAKING_KEY_PATH = [
+  PURPOSE,
+  COIN_TYPE,
+  ACCOUNT_INDEX + HARD_DERIVATION_START,
+  CHIMERIC_ACCOUNT,
+  STAKING_KEY_INDEX,
+]
+export const BACKEND = {
+  API_ROOT,
+  TOKEN_INFO_SERVICE,
+  FETCH_UTXOS_MAX_ADDRESSES: 50,
+  TX_HISTORY_MAX_ADDRESSES: 50,
+  FILTER_USED_MAX_ADDRESSES: 50,
+  TX_HISTORY_RESPONSE_LIMIT: 50,
+} as const
+export const CHAIN_NETWORK_ID = 0
+export const BIP44_DERIVATION_LEVELS = {
+  ROOT: 0,
+  PURPOSE: 1,
+  COIN_TYPE: 2,
+  ACCOUNT: 3,
+  CHAIN: 4,
+  ADDRESS: 5,
+} as const
+export const REWARD_ADDRESS_ADDRESSING = {
+  path: [PURPOSE, COIN_TYPE, ACCOUNT_INDEX + HARD_DERIVATION_START, CHIMERIC_ACCOUNT, STAKING_KEY_INDEX],
+  startLevel: BIP44_DERIVATION_LEVELS.PURPOSE,
+}
+export const KEY_DEPOSIT = '2000000'
+export const POOL_DEPOSIT = '500000000'
+export const LINEAR_FEE = {
+  COEFFICIENT: '44',
+  CONSTANT: '155381',
+} as const
+export const MINIMUM_UTXO_VAL = '1000000'
+
+export const IS_MAINNET = false
+
+export const PROTOCOL_MAGIC = 1097911063
```

Comment:
Probably it's better to split into testnet preview and testnet preprod. For preprod protocol magic is 1 for preview is 2

Reply:
I'm splitting Byron, Shelley, Shelley testnet apart so the values are static and not dynamic. do the numbers look correct?

Reply:
I think by following the logic applied it should be `/shelley-{preprod,preview}/`